doc: add Repo to the Python API reference manual.

rholy opened the pull request.

jsilhan commented on doc/reference-py-repo.rst:
```diff
@@ -0,0 +1,78 @@
+.. class:: hawkey.Repo
+
+  Instances of :class:`hawkey.Repo` point to metadata of packages that are
+  available to be installed. The metadata are expected to be in the format
```
...to be in the *XML* format...
Createrepo can generate .sqlite files too, readable by yum and not by libsolv

jsilhan commented on doc/reference-py-repo.rst:
```diff
@@ -0,0 +1,78 @@
+    An integer priority value of this repository. If there is more than one
+    candidate package for a particular operation, the one from the repository
+    with *the lowest priority* value is picked, possibly despite being less
+    convenient otherwise (e.g. by being a lower version).
```
So `cost` is taken into consideration only if two packages are at the same version meanwhile `priority` ignores versions?

rholy replied:
No, both ignores the version. It works this way:
1. find the repositories containing a package that matches the ```spec```
2. then select just those repositories that have the lowest priority
3. then select those priorities that have the lowest cost

I just copy&pasted the description from: http://dnf.readthedocs.org/en/latest/conf_ref.html#repo-options
Libsolv contains in its documentation: "The sub-priority of the repository. This value is compared when the priorities
of two repositories are the same. It is useful to make the library prefer
on-disk repositories to remote ones." (please note that we implement ```cost``` using libsolv's ```subpriority```).
I did not mention ```priority``` in the description of ```cost``` because it's not mention in DNF's documentation. But if you think it's worth, I can.

jsilhan commented:
Mention it in both docs, please. It would be helpful for user - I was confused.

rholy commented:
I've adjusted the ```hawkey.Repo``` class description and the description of the metadata attributes to match the documentation of libsolv.
I've changed the description of ```hawkey.Repo.cost```.
It should be ready to be merged.

rholy commented:
I've adjusted the ```hawkey.Repo``` class description and the description of the metadata attributes to match the documentation of libsolv.
I've changed the description of ```hawkey.Repo.cost```.
It should be ready to be merged.

jsilhan commented:
Merged, thanks
